articles/100_ament.md

```diff
@@ -0,0 +1,102 @@
+  * changing them or adding further features was "challenging"
+* ament uses extension points to allow other packages to hook into the process as necessary
+* in ament the tools are strictly separated, modification to the tools do not require a release of ament itself
+
+
+## How does *ament* relate to *catkin*?
+
+`catkin` has been used for ROS 1 since ROS Groovy.
+`catkin` is based around CMake and even packages only containing Python code are being processed via CMake.
+It was developed as a replacement of `rosbuild` which was used from the beginning of ROS.
+Therefore various features of catkin have been designed to be similar to rosbuild.
+
+
+### Why not continue to use *catkin*?
+
+catkin has many advantages over rosbuild.
+Those cover out-of-source builds, automatic CMake config file generation, an installation target and many more.
+But for some cases it is not flexible enough or not as easy to use as it should be.
+A few of these cases are mentioned below:
+
+
+#### *Devel space*
+
+catkin has the feature to provide a so called *devel space* after building a set of packages.
+That folder is providing a fully working ROS environment without the need to install the packages.
+Avoiding to copy any files allows to e.g. edit Python code and immediately try running the code.
+
+While this is a very convenient feature and speeds up the development process it comes at a cost.
+The additional logic in catkin increased its complexity significantly.
+Additionally the CMake code in every ROS package had to make sure to handle the *devel space* correctly which puts an extra effort on every ROS developer.
+
+`ament` provides the same advantage using the option of *symlinked installs* without the extra complexity for each ROS package.
+
+
+#### catkin_simple
+
+The ROS package `catkin_simple` was the attempt to make the common cases of developing ROS packages easier.
+While it is able to reduce the complexity of the CMake code in some ROS package it fails conceptually in other cases.
+Some of the limitations were due to core catkin design decisions like the order and position of calling certain CMake functions etc.
+
+E.g. the function `catkin_package()` must be invoked *before* any target in order to setup the appropriate location for the build targets in the devel space.
+But in order to automatically perform several tasks in the code generation step this functionality needs to happen *after* all target have been defined.
+
+
+#### Building within a single CMake context
+
+catkin can allow build multiple packages within a single CMake context (using `catkin_make`).
+While this significantly speeds up the process it is not scalable at large.
+This is due to the fact that all packages share the same CMake namespace.
+Therefore the package might have colliding target names or fail to build correctly due to side effects between the packages.
```

Comment:
I would think the point is covered in line 42:

    `catkin` is based around CMake and even packages only containing Python code are being processed via CMake.